pkg/liqo-controller-manager/virtualNode-controller/namespaceMap_lifecycle.go

```diff
@@ -0,0 +1,115 @@
+package virtualNode_controller
+
+import (
+	"context"
+	"fmt"
+	mapsv1alpha1 "github.com/liqotech/liqo/apis/virtualKubelet/v1alpha1"
+	const_ctrl "github.com/liqotech/liqo/pkg/liqo-controller-manager"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/util/slice"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// remove Finalizer and Update the NamespaceMap
+func (r *VirtualNodeReconciler) deleteNamespaceMap(nm mapsv1alpha1.NamespaceMap) error {
+
+	nm.SetFinalizers(slice.RemoveString(nm.GetFinalizers(), namespaceMapFinalizer, nil))
+	klog.Info("Someone try to delete namespaceMap, ok delete!!")
+
+	if err := r.Update(context.TODO(), &nm); err != nil {
+		klog.Error(err, " --> Problems while removing NamespaceMap's finalizer")
+		return err
+	}
+	return nil
+}
+
+// create a new NamespaceMap with Finalizer and OwnerReference
+func (r *VirtualNodeReconciler) createNamespaceMap(n corev1.Node) error {
+	klog.Infof(" Create NamespaceMap for %s", n.Name)
+	nm := &mapsv1alpha1.NamespaceMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "namespaceresources.liqo.io/v1",
+			Kind:       "NamespaceMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: fmt.Sprintf("%s-", n.GetAnnotations()[const_ctrl.VirtualNodeClusterId]),
+			Namespace:    const_ctrl.MapNamespaceName,
+			Labels: map[string]string{
+				const_ctrl.VirtualNodeClusterId: n.GetAnnotations()[const_ctrl.VirtualNodeClusterId],
+			},
+		},
+	}
+
+	nm.SetFinalizers(append(nm.GetFinalizers(), namespaceMapFinalizer))
+
+	nm.SetOwnerReferences(append(nm.OwnerReferences, metav1.OwnerReference{
```

Comment:
You can use the functions provided by the controller-runtime ( https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#SetControllerReference ) to do so

Reply:
thank you for the advise